pylint/checkers/base/basic_checker.py

```diff
@@ -657,13 +657,38 @@ def _check_misplaced_format_function(self, call_node: nodes.Call) -> None:
             ):
                 self.add_message("misplaced-format-function", node=call_node)
 
+    @staticmethod
+    def _is_terminating_func(node: nodes.Call) -> bool:
+        """Detect call to exit(), quit(), os._exit(), or sys.exit()."""
+        if (
+            not isinstance(node.func, nodes.Attribute)
+            and not (isinstance(node.func, nodes.Name))
+            or isinstance(node.parent, nodes.Lambda)
+        ):
+            return False
+
+        qnames = {"_sitebuiltins.Quitter", "sys.exit", "posix._exit", "nt._exit"}
+
+        try:
+            for inferred in node.func.infer():
+                if inferred.qname() in qnames:
+                    return True
+        except Exception:  # pylint: disable=broad-except
```

Comment:
This shouldn't be `Exception` but only catch stuff we actually want to catch here. The `pylint` warning exists for a reason 😄

Reply:
> Let's just use the try...except for all of the inference. 

then the code base actually has some uses of `except Exception:  # pylint: disable=broad-except`.

Please beware of your conflicting reviews. It's really frustrating for contributors.

Reply:
I don't understand. We know which exceptions we want to catch here right? `StopIteration`, `InferenceError`? We shouldn't be using a broad except here.